go/libkb/httpsrv.go

```diff
@@ -125,6 +124,19 @@ func (h *HTTPSrv) Start() (err error) {
 	return nil
 }
 
+// EnsureActive is like Start, except that in case the server is already active
+// at the time this is called, it returns a nil error rather than
+// errHTTPServerAlreadyRunning.
+func (h *HTTPSrv) EnsureActive() (err error) {
```

Comment:
You could override `Start` to do the right thing.

Reply:
It's also used in chat, and I was not sure if Mike is OK with changing. But I guess I'll push a follow-up commit and see what Mike thinks. We can also just move this type into libhttpserver.